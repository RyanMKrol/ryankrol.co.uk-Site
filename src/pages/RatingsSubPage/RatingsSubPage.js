import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import { DataCollection, MovieRatingsItem, AlbumRatingsItem } from '../../components';
import styles from './RatingsSubPage.module.css';

import { baseEndpoint, generateCustomSorter } from '../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class RatingsSubPage extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.moviesEndpoint = `${baseEndpoint()}/ratings/movie`;
    this.albumsEndpoint = `${baseEndpoint()}/ratings/album`;
    this.moviesOutputLimit = 6;
    this.albumsOutputLimit = 6;
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h2>Movies I've Watched</h2>
        <br />
        <DataCollection
          outputSize={this.moviesOutputLimit}
          endpoint={this.moviesEndpoint}
          itemTag={MovieRatingsItem}
          sorter={generateCustomSorter('blind')}
        />
        <Link className={styles.seeMore} to={'/MovieRatings'}>
          <h2>...</h2>
        </Link>

        <h2>Albums I've Listened To</h2>
        <br />
        <DataCollection
          outputSize={this.albumsOutputLimit}
          endpoint={this.albumsEndpoint}
          itemTag={AlbumRatingsItem}
          sorter={generateCustomSorter('rating')}
        />
        <Link className={styles.seeMore} to={'/AlbumRatings'}>
          <h2>...</h2>
        </Link>
      </div>
    );
  }
}

export default RatingsSubPage;
