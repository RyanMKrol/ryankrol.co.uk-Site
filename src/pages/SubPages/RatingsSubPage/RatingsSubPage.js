import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import { DataCollection, MovieRatingsItem, AlbumRatingsItem } from '../../../components';
import styles from './RatingsSubPage.module.css';

import { baseEndpoint, generateCustomSorter } from '../../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class RatingsSubPage extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.moviesEndpoint = `${baseEndpoint()}/ratings/movie`;
    this.albumsEndpoint = `${baseEndpoint()}/ratings/album`;
    this.moviesOutputLimit = 6;
    this.albumsOutputLimit = 6;
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h2>Movies I've Watched</h2>
        <p>
          Like most humans, I enjoy watching movies! Unlike most humans however, I've taken the time
          to setup a system to track what I think of the movies I watch. If you'd like to see how
          I've rated anything I've watched since December 2020, then look no further!
        </p>
        <br />
        <DataCollection
          outputSize={this.moviesOutputLimit}
          endpoint={this.moviesEndpoint}
          itemTag={MovieRatingsItem}
          sorter={generateCustomSorter('blind')}
        />
        <Link className={styles.seeMore} to={'/MovieRatings'}>
          <h2>...</h2>
        </Link>
        <h2>Albums I've Listened To</h2>
        <p>Like the movie ratings, but this time, with music!</p>
        <br />
        <DataCollection
          outputSize={this.albumsOutputLimit}
          endpoint={this.albumsEndpoint}
          itemTag={AlbumRatingsItem}
          sorter={generateCustomSorter('rating')}
        />
        <Link className={styles.seeMore} to={'/AlbumRatings'}>
          <h2>...</h2>
        </Link>
      </div>
    );
  }
}

export default RatingsSubPage;
