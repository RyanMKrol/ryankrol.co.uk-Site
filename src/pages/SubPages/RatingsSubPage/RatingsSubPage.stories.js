import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import {
  FULL_SIZE_ALBUMS_MOCK,
  LIMITED_SIZE_ALBUMS_MOCK,
  FULL_SIZE_MOVIES_MOCK,
  LIMITED_SIZE_MOVIES_MOCK,
} from './__mocks__/RatingsSubPageMocks';

export default {
  title: 'Pages/RatingsSubPage',
};

export const FullSizeAlbumRatings = () => <Router>{FULL_SIZE_ALBUMS_MOCK}</Router>;
export const LimitedSizeAlbumRatings = () => <Router>{LIMITED_SIZE_ALBUMS_MOCK}</Router>;
export const FullSizeMovieRatings = () => <Router>{FULL_SIZE_MOVIES_MOCK}</Router>;
export const LimitedSizeMovieRatings = () => <Router>{LIMITED_SIZE_MOVIES_MOCK}</Router>;
