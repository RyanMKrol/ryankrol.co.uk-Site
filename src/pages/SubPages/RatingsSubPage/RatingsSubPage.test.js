import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import RatingsSubPage from './RatingsSubPage';

it('renders playground correctly', () => {
  const header = shallow(<RatingsSubPage />);
  expect(toJson(header)).toMatchSnapshot();
});
