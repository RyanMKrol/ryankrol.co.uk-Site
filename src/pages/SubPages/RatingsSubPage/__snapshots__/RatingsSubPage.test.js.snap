// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RatingsSubPage correctly 1`] = `
<div
  className="container"
>
  <h2>
    Movies I've Watched
  </h2>
  <p>
    Like most humans, I enjoy watching movies! Unlike most humans however, I've taken the time to setup a system to track what I think of the movies I watch. If you'd like to see how I've rated anything I've watched since December 2020, then look no further!
  </p>
  <br />
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/ratings/movie"
    itemTag={[Function]}
    outputSize={6}
    sorter={[Function]}
  />
  <Link
    className="seeMore"
    to="/MovieRatings"
  >
    <h2>
      ...
    </h2>
  </Link>
  <h2>
    Albums I've Listened To
  </h2>
  <p>
    Like the movie ratings, but this time, with music!
  </p>
  <br />
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/ratings/album"
    itemTag={[Function]}
    outputSize={6}
    sorter={[Function]}
  />
  <Link
    className="seeMore"
    to="/AlbumRatings"
  >
    <h2>
      ...
    </h2>
  </Link>
</div>
`;
